src/compiler/checker.ts

```diff
@@ -5742,6 +5752,9 @@ namespace ts {
                         prop.syntheticOrigin = propertySymbol;
                         prop.declarations = propertySymbol.declarations;
                     }
+                    else if (!isIdentifierText((<StringLiteralType>t).value, compilerOptions.target)) {
+                        prop.syntheticName = `"${escapeString((<StringLiteralType>t).value, CharacterCodes.doubleQuote)}"`;

```

Comment:
why use `escapeString` instead of `escapeLeadingUnderscores` as in `propName`?

Reply:
What is `propName` in the buggy case, anyway? Why not change `propName` in this case to be the version that's called with `escapeString`, because then line 5748 (`const prop = createSymbol(...`) would do this name lookup that we need for free.